applications/DEMApplication/python_scripts/DEM_analysis_stage.py

```diff
@@ -465,6 +465,10 @@ def UpdateIsTimeToPrintInOneModelPart(self, model_part, is_time_to_print):
     def BeforePrintingOperations(self, time):
         pass
 
+    def PrintAnalysisStageProgressInformation(self):
+        if self.DEM_parameters["echo_level"].GetInt() > 1:
```

Comment:
As you say, the DEM is a special case, with many many time steps. For that, we created the 'report' long ago. It was being used by the old main_script.py.
https://github.com/KratosMultiphysics/Kratos/blob/b99024dfdc88f51af1e03fd672a668f3413ea655/applications/DEMApplication/python_scripts/main_script.py#L449
Can we call that object instead of putting this `if`?
We could not do it before because the AnalysisStage did not have this overridable method...

Reply:
For help, you can find self.report in several spots inside `main_script.py`.

Reply:
Hi, I just checked it and realized that PrintAnalysisStageProgressInformation is called in the InitializeSolutionStep of the analysis_stage base class, while StepiReport is called in the FinalizeSolutionStep... I think we should only call this once per step. Where do you prefer it?

Reply:
We should do it in the same fashion as the rest of Kratos. Then, inside `PrintAnalysisStageProgressInformation  `is fine, so it will be prompted as part of `InitializeSolutionStep `.

Reply:
Done